handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1943,14 +1955,36 @@ public void run() {
      * @return {@code true} if {@link #handshakePromise} was set successfully and a {@link SslHandshakeCompletionEvent}
      * was fired. {@code false} otherwise.
      */
-    private boolean setHandshakeSuccess() {
+    private boolean setHandshakeSuccess() throws SSLException {
         // Our control flow may invoke this method multiple times for a single FINISHED event. For example
         // wrapNonAppData may drain pendingUnencryptedWrites in wrap which transitions to handshake from FINISHED to
         // NOT_HANDSHAKING which invokes setHandshakeSuccess, and then wrapNonAppData also directly invokes this method.
+        SSLSession session = engine.getSession();
+        if (resumptionController != null && session.getPeerPrincipal() instanceof X500Principal) {
```

Comment:
Looks like we can't make this instanceof check, but have to accept that "peer not verified" may be thrown